### PR TITLE
add connecting with ledger UI

### PR DESCRIPTION
**Pull request by acharb**
https://user-images.githubusercontent.com/30449853/173958301-f730261e-d47c-4445-9ce2-0b7bcde31a1d.mov

### CODE REVIEW

File: extension/src/popup/views/Account/styles.scss [hunk 1]
```diff
@@ -1,9 +1,9 @@
 :root {
-  --account-view-padding-left: 1.5rem;
+  --account-view-padding-side: 1.5rem;
```
Comment (acharb) [inline extension/src/popup/views/Account/styles.scss:2]: took me a solid min to figure out why we were using only `left` for the side padding 😆  so I changed to `side`

File: extension/src/popup/components/hardwareConnect/LedgerConnect/index.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  if (message.indexOf("Incorrect length")) {
+    return LEDGER_ERROR.NOT_OPEN;
+  }
+  return LEDGER_ERROR.OTHER;
+};
+
+export const LedgerConnect = ({ goBack }: { goBack?: () => void }) => {
+  const [isConnecting, setIsConnecting] = useState(false);
+
+  // TODO - move to redux with bipPath
```
Comment (acharb) [inline extension/src/popup/components/hardwareConnect/LedgerConnect/index.tsx:40]: will most probably move the connect stuff into redux, wasn't sure of the best API yet so doing in later PR

File: extension/package-lock.json [hunk 3]
```diff
@@ -1,13 +0,0 @@
-{
```
Comment (acharb) [inline extension/package-lock.json:1]: this isn't needed right?

File: extension/src/popup/ducks/accountServices.ts [hunk 4]
```diff
@@ -238,15 +238,17 @@ interface InitialState {
   applicationState: APPLICATION_STATE;
   hasPrivateKey: boolean;
   publicKey: string;
+  connectingWalletType: string;
```
Comment (piyalbasu) [inline extension/src/popup/ducks/accountServices.ts:241]: pretty small - but should the type here be `WalletType`? Would make it clear that we're looking for something from that enum here

File: extension/src/popup/components/hardwareConnect/LedgerConnect/index.tsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  const [connectError, setConnectError] = useState("");
+
+  const handleConnect = async () => {
+    setIsConnecting(true);
+    try {
+      const transport = await TransportWebUSB.request();
+      const ledgerApi = new LedgerApi(transport);
+      const response = await ledgerApi.getPublicKey(defaultStellarBipPath);
+
+      setPublicKey(response.publicKey);
```
Comment (piyalbasu) [inline extension/src/popup/components/hardwareConnect/LedgerConnect/index.tsx:52]: Awesome that their API looks pretty easy to use!

**Review by piyalbasu — APPROVED** [review]

File: extension/package-lock.json (hunk 3 above)
Comment (piyalbasu) [reply extension/package-lock.json:1]: Yup! Can be deleted, thanks!

File: extension/src/popup/ducks/accountServices.ts (hunk 4 above)
Comment (acharb) [reply extension/src/popup/ducks/accountServices.ts]: ya  good call https://github.com/stellar/freighter/pull/531/commits/b1bba6d7b161390d23d023bc7ad0fd87021d1dec

**Pull request merged**